### PR TITLE
Update argument names of set methods to use element instead of x

### PR DESCRIPTION
**Pull request by bmcdonald3**
In the set module review, it was decided that we would prefer the name of `element` for argument names in set methods, where previously the argument names were just `x`. This PR deprecates the old methods and adds new ones with the new argument names, so anyone calling the methods with the `x` argument by name will get the deprecation message. Methods changed: `add()`, `contains()`, `remove()`.

- [x] paratest

Closes https://github.com/chapel-lang/chapel/issues/18651

### CODE REVIEW

**Comment by cassella** [comment]
Are there deprecation versions coming for users passing the arguments by the old name?

File: modules/standard/Set.chpl [hunk 1]
```diff
@@ -266,15 +266,22 @@ module Set {
...
+      // Remove `on this` block because it prevents copy elision of `element`
+      // when passed to `_addElem`. See #15808.
+      _enter(); defer _leave();
+      _addElem(element);
+    }
+
+    pragma "last resort"
+    deprecated "The argument name `x` has been deprecated for function `add`, please use `element` instead"
+    proc ref add(in x: eltType) lifetime this < x {
       _enter(); defer _leave();
```
Comment (lydia-duncan) [inline modules/standard/Set.chpl:285]: The lifetime constraint may mean this doesn't work, but I tend to like having the deprecated function call the new function as much as possible.  This code is probably short enough that it is less likely but I worry about someone else coming in and modifying the wrong version of the function, and then having the changes not be reflected in the new version when the old version gets deleted

**Review by lydia-duncan — APPROVED** [review]
Just a little nit that it is okay to ignore (especially if it isn't possible), but figured I should bring it up.  Otherwise, this is good to go!

File: modules/standard/Set.chpl (hunk 1 above)
Comment (bmcdonald3) [reply modules/standard/Set.chpl]: It looks like that works just fine, I've changed the code to reflect that; thanks for the suggestion.

**Pull request merged**